src/Datadog.Trace/ExtensionMethods/HeadersCollectionExtensions.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Globalization;
+using Datadog.Trace.Headers;
+
+namespace Datadog.Trace.ExtensionMethods
+{
+    /// <summary>
+    /// Extension methods for <see cref="IHeadersCollection"/>.
+    /// </summary>
+    public static class HeadersCollectionExtensions
+    {
+        private const NumberStyles NumberStyles = System.Globalization.NumberStyles.Integer;
+        private static readonly CultureInfo InvariantCulture = CultureInfo.InvariantCulture;
+
+        /// <summary>
+        /// Creates a <see cref="SpanContext"/> from the values found in the specified headers.
+        /// </summary>
+        /// <param name="headers">The headers that contain the values to be extracted.</param>
+        /// <returns>A new <see cref="SpanContext"/> that contains values extracted from <paramref name="headers"/>.</returns>
+        public static SpanContext ExtractSpanContext(this IHeadersCollection headers)
+        {
+            if (headers == null)
+            {
+                throw new ArgumentNullException(nameof(headers));
+            }
+
+            ulong traceId = ParseUInt64(headers.GetValues(HttpHeaderNames.TraceId));
```

Comment:
We default these values to `0` if we cannot parse them, is there any place where we create a new trace id if the one in the `SpanContext` is `0`?

Or some place where we check that the trace id is not `0` before activating this `SpanContext`

Reply:
nvm, I see below you updated `SpanContext` with `parent`. However, we might be better off doing something like:

```dotnet
if traceId == 0 {
    return null
}

return new SpanContext(traceId, parentId, (SamplingPriority)samplingPriority);
```

This way there is no way we can propagate an invalid SpanContext... there will be that one place we don't check `SpanContext.TraceId > 0`

Reply:
Thanks! Added in db215e46595512c6c211a3ec0ee29854a2f7aa0b.

Reply:
Great!